tfx_addons/message_exit_handler/proto/slack.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package tfx_addons.slack_exit_handler;
+
+message SlackSpec {
+  // Sets the slack credentials.
+
+  // Slack token: secret to authenticate with slack.
+  string slack_token = 1;
+
+  // Slack channel: channel/user_id to post to.
+  string slack_channel_id = 2;
+
+}
```

Comment:
I am open to pydantic, etc. It just followed the TFX design patterns from the BulkInferrer, InfraValidator, etc. I think the benefit is that the information can be serialized as passed to the component as a string instead of passing every argument individually.

What non-proto structure would you recommend? Plain `json`?

Reply:
pydantic accepts serializing from/to json so yes.

Reply:
@casassg Let's turn the feedback into implementation guidelines for the repo. I think those considerations are good to consider.

Reply:
Agreed. But this is a port from an internal package and we used protos heavily at Digits (a bunch of folks at Digits worked at Google previously). I ported everything  to Pydantic. I think we should collect a few more of those guidelines and add them to the readme.

Reply:
Switched to Pydantic